cpp/tests/cppflow.cc

```diff
@@ -0,0 +1,100 @@
+#include <algorithm>
+#include <exception>
+#include <functional>
+#include <iostream>
+#include <random>
+#include <vector>
+#include <ctime>
+#include <catch.hpp>
+
+#include <sopt/imaging_forward_backward.h>
+#include <sopt/logging.h>
+#include <sopt/maths.h>
+#include <sopt/relative_variation.h>
+#include <sopt/sampling.h>
+#include <sopt/types.h>
+#include <sopt/utilities.h>
+#include <sopt/wavelets.h>
+#include <cppflow/cppflow.h>
+#include "cppflow/ops.h"
+#include "cppflow/model.h"
+
+// This header is not part of the installed sopt interface
+// It is only present in tests
+#include <tools_for_tests/directories.h>
+#include <tools_for_tests/tiffwrappers.h>
+
+// \min_{x} ||\Psi^Tx||_1 \quad \mbox{s.t.} \quad ||y - Ax||_2 < \epsilon and x \geq 0
+
+typedef double Scalar;
+typedef sopt::Vector<Scalar> Vector;
+typedef sopt::Matrix<Scalar> Matrix;
+typedef sopt::Image<Scalar> Image;
+
+TEST_CASE("Cppflow"){
+
+  std::string const input_image = "cameraman256";
+  Image const image = sopt::notinstalled::read_standard_tiff(input_image);
+
+  //TODO use image.shape nad image.size instead of hardcoding values
+  //TODO check data type and allow float or double
+  
+  // create a vector
+  std::cout << "============Create vector" << std::endl;
+  int const image_rows = image.rows();
+  int const image_cols = image.cols();
+  int const image_size = image.size();
+
+  std::vector<int64_t> tensor_shape = {image_rows, image_cols};
```

Comment:
Not sure about this, but could we just hard code the extra dimensions in here?
```suggestion
  std::vector<int64_t> tensor_shape = {1, image_rows, image_cols, 1};
```

Reply:
Yeah that works! Great, makes it look a bit neater